thoth/common/openshift.py

```diff
@@ -377,21 +377,35 @@ def get_pod_log(
 
         return response.text
 
-    def get_workflow_node_log(
+    def get_workflow_pod_name(
         self, node_name: str, workflow_id: str, namespace: str
     ) -> Optional[str]:
-        """Get log from a task/node in a workflow."""
+        """Get pod name where task is being executed."""
         workflow = self.get_workflow(workflow_id, namespace=namespace)
         nodes = workflow.get("status", {}).get("nodes", {})
 
         for pod_name, node_info in nodes.items():
             if node_info["displayName"] == node_name:
-                return self.get_pod_log(pod_name, namespace=namespace, container="main")
+                return pod_name
```

Comment:
@fridex I'm getting a lot of pre-commit errors:
```
thoth/common/openshift.py:389:17: error: Returning Any from function declared to return "Optional[str]"  [no-any-return]
thoth/common/openshift.py:400:33: error: Argument 1 to "get_pod_log" of "OpenShift" has incompatible type "Optional[str]"; expected "str"  [arg-type]
thoth/common/openshift.py:407:16: error: Incompatible return value type (got "Dict[str, Any]", expected "Optional[str]")  [return-value]
thoth/common/openshift.py:407:36: error: Argument 1 to "get_pod_status" of "OpenShift" has incompatible type "Optional[str]"; expected "str"  [arg-type]
Found 4 errors in 1 file (checked 13 source files)
```
So, I tried returning `None` and moving 
```
raise NotFoundException(
     f"No node {node_name!r} in workflow {workflow_id!r} in namespace {namespace!r} found"
)
```
to `get_workflow_node_log` and `get_workflow_node_status`, but still not working, so I don't get why `pre-commit` is unhappy :smile: ?

Reply:
```
thoth/common/openshift.py:389:17: error: Returning Any from function declared to return "Optional[str]"  [no-any-return]
```

You are iterating over nodes but the type of `pod_name` is not known (hence type `Any`).

```
thoth/common/openshift.py:400:33: error: Argument 1 to "get_pod_log" of "OpenShift" has incompatible type "Optional[str]"; expected "str"  [arg-type]
thoth/common/openshift.py:407:16: error: Incompatible return value type (got "Dict[str, Any]", expected "Optional[str]")  [return-value]
thoth/common/openshift.py:407:36: error: Argument 1 to "get_pod_status" of "OpenShift" has incompatible type "Optional[str]"; expected "str"  [arg-type]
```

You can adjust the return type of function to be directly `str` instead of `Optional[str]` as it always return the `pod_name` and raises otherwise.